src/data/sym/card.lean

```diff
@@ -105,10 +104,21 @@ begin
   exact and_iff_right ⟨a, mem_univ _, ha⟩,
 end
 
-protected lemma card [fintype α] :
-  card (sym2 α) = card α * (card α + 1) / 2 :=
-by rw [←fintype.card_congr (@equiv.sum_compl _ is_diag (sym2.is_diag.decidable_pred α)),
-  fintype.card_sum, card_subtype_diag, card_subtype_not_diag, nat.choose_two_right, add_comm,
-  ←nat.triangle_succ, nat.succ_sub_one, mul_comm]
+/-- Finset **stars and bars** for the case `n = 2`. -/
+lemma _root_.finset.card_sym2 (s : finset α) : s.sym2.card = s.card * (s.card + 1) / 2 :=
+begin
+  rw [←image_diag_union_image_off_diag, card_union_eq, sym2.card_image_diag,
+    sym2.card_image_off_diag, nat.choose_two_right, add_comm, ←nat.triangle_succ, nat.succ_sub_one,
+    mul_comm],
+  rintro m he,
+  rw [inf_eq_inter, mem_inter, mem_image, mem_image] at he,
+  obtain ⟨⟨a, ha, rfl⟩, b, hb, hab⟩ := he,
+  refine not_is_diag_mk_of_mem_off_diag hb _,
+  rw hab,
+  exact is_diag_mk_of_mem_diag ha,
+end
```

Comment:
Do you have any intuition for why this proof is longer than the type version? Is this intrinsic complexity, or are there some additional lemmas that would be useful?

Reply:
In the type version, I had the statement that `s.diag.image quotient.mk` and `s.off_diag.image quotient.mk` were disjoint already thanks to `sym2.card_subtype_diag` and `sym2.card_subtype_not_diag`. Those lemmas don't make sense to have about finsets, so I inlined them, and the overall complexity is slightly lower.

Reply:
I can add the disjointness of `s.diag.image quotient.mk` and `s.off_diag.image quotient.mk` as as separate lemma if you want, but to me that's too specialized.

Reply:
I don't see another obvious way to split this proof either, so let's keep it as it is.